src/client.js

```diff
@@ -13,9 +13,11 @@ const createDecipher = require('./transforms/encryption').createDecipher
 const closeTimeout = 30 * 1000
 
 class Client extends EventEmitter {
-  constructor (isServer, version, customPackets, hideErrors = false) {
+  constructor (isServer, version, customPackets, hideErrors = false, customCommunication = undefined) {
```

Comment:
Hm, pretty sure it would be better if it was an object like
```suggestion
  constructor (isServer, version, additionalOptions) {
```

Reply:
I don't think this whole change makes a lot of sense.

What about instead letting the user provide its own Client class ?

This is overriding pretty much everything so I don't think it really fits in Client